added lobby player list and change host for hosts

mliu59 opened the pull request:
Added 3 endpoints for lobby API (only accessible by host):
- get all users in a lobby
- get all anon_users in a lobby
- change host to another user in the lobby

Added corresponding UI section for lobby hosts:
https://vimeo.com/727473707/31e5137af3

bricehalder commented on templates/lobbys/lobby.html:
```diff
@@ -177,10 +211,45 @@ <h2 v-else>Lobby [[lobbyInfo.lobby_id]]: </h2>
+                    <table class="table table-hover">
+                        <thead>
+                            <th scope="col">Display name</th>
+                            <th scope="col">Host</th>
```
Is the host column necessary? I can see this being useful if we want to support multiple hosts in the future, but right now it seems like we just support one, in which case I think the column is confusing to have in the UI (since only the host sees this anyways).

Related - I think "Make host" is not clear right now, since with the Host column, it feels like you are _adding_ a new host rather than _transferring_ ownership. I think we should either 
1. change the functionality here to add an additional host
  1a. Change the backend/db schema to support a set of userid's rather than a single userid
2. change the text to reflect the transfer i.e. "Make host" --> "Transfer Ownership" or similar and make that clear in the pop-up as well.

bricehalder commented on templates/lobbys/lobby.html:
```diff
@@ -177,10 +211,45 @@ <h2 v-else>Lobby [[lobbyInfo.lobby_id]]: </h2>
+                                <td>[[user.username]]</td>
+                                <td><i v-if="user.owner" class="bi bi-check"></i></td>
+                                <td><i class="bi bi-check"></i></td>
+                                <td><button v-if="!user.owner" class="btn btn-link btn-sm" @click="makeHost(user.user_id, user.username)"><small>Make host</small></button></td>
```
See above comment regarding "Make host" text

But in addition:

(Personal preference) 
I think this should probably be a dropdown menu with options per user (for now, this can be the only thing in the dropdown).

1. It would be cleaner UI
2. We probably want to support more user management options in the future, namely kicking or banning users from the lobby, and this would be a clean and extensible way to support those additional options per user.

bricehalder commented:
Nice, I think the functional part looks good. Left some comments mostly around clarity of our UI

bricehalder commented on templates/lobbys/lobby.html:
```diff
@@ -177,10 +211,45 @@ <h2 v-else>Lobby [[lobbyInfo.lobby_id]]: </h2>
+                        <thead>
+                            <th scope="col">Display name</th>
+                            <th scope="col">Host</th>
+                            <th scope="col">Full user</th>
```
(personal preference)

I think "Full user" is kind of confusing too. I think "Registered User" or "Anonymous User" (and flipping the condition) would be more clear

dqian3 commented on templates/lobbys/lobby.html:
```diff
@@ -150,6 +157,33 @@
+
+
+        async getLobbyUsers() {
+            const resp = await fetchJson(`/api/lobbys/players/${LOBBY_ID}`, "GET");
```
Small thing, but these can be done in parallel using Promise.all().

mliu59 commented:
Addressed comments, see image
![image](https://user-images.githubusercontent.com/61032765/178625310-b5ed1c43-b4fb-44cc-a6a1-52691f95b31b.png)

The dropdown implementation is fairly crude, but bootstrap was giving me alot of issues with aria attributes, so I opted for a workaround

bricehalder commented on templates/lobbys/lobby.html:
```diff
@@ -177,10 +226,58 @@ <h2 v-else>Lobby [[lobbyInfo.lobby_id]]: </h2>
+                                        <div class="dropdown-menu" 
+                                            v-bind:id="String(user.user_id)+'dropdown'">
+                                            <button v-if="!user.owner" class="btn btn-sm" @click="makeHost(user.user_id, user.username)">Transfer Ownership</button>
+                                            <button class="btn btn-sm disabled" v-else>Currently host</button>
```
(personal preference)
Don't really think the "Currently host" is needed since it's implied, so I would probably just delete this line. I was kind of confused looking at the example image.

bricehalder reviewed (approved):
Looks good, just another small comment
